File: packages/platform/src/ipc/publish-log.ts
/**
 * 发布日志 IPC 封装 — 日志分页查询 + 清空。
 *
 * 后端：壳层 `commands/publish_log.rs`（InMemoryPublishLogStore + app::publish::PublishLogService）。
 *
 * @author agent
 * @created 2026-08-13
 */

import { call } from "./invoke";

/** 发布日志条目（与 Rust `PublishLog` 对齐核心字段）。 */
export interface PublishLog {
  id: number;
  owner_id: number;
  account_id: string;
  title: string;
  price: string | null;
  status: "pending" | "publishing" | "success" | "failed";
  item_url: string | null;
  item_id: string | null;
  error_message: string | null;
  resolved_address_text: string | null;
  address_source: string | null;
  created_at: string | null;
}

const OWNER_ID = 1; // 桌面单用户；多用户时由登录态注入

/** 分页查询发布日志。 */
export function publishLogList(query: {
  page: number;
  page_size: number;
  account_id?: string;
  status?: string;
}): Promise<[PublishLog[], number]> {
  return call<[PublishLog[], number]>("publish_log_list", {
    request: {
      owner_id: OWNER_ID,
      page: query.page,
      page_size: query.page_size,
      account_id: query.account_id ?? "",
      status: query.status ?? "",
    },
  });
}

/** 清空 N 天前的日志（0 = 全部）。 */
export function publishLogClear(days = 10): Promise<void> {
  return call<void>("publish_log_clear", {
    request: { owner_id: OWNER_ID, days },
  });
}

File: apps/desktop/src-tauri/src/platforms/xianyu/ipc/publish_log.rs
//! 发布日志 Tauri commands — 分页查询与按天数清理。
//!
//! 壳层组合：`InMemoryPublishLogStore` → `app::publish::PublishLogService`。

use crate::platforms::xianyu::persist::InMemoryPublishLogStore;
use app::publish::{PublishLog, PublishLogQuery, PublishLogService};
use common;
use serde::Deserialize;
use std::sync::Arc;
use tauri::State;

/// 日志服务句柄（setup 时注册到 Tauri 状态）。
pub struct PublishLogHandle {
    pub store: Arc<InMemoryPublishLogStore>,
}

/// 日志查询请求。
#[derive(Debug, Deserialize)]
pub struct PublishLogListRequest {
    pub owner_id: i64,
    pub page: u32,
    pub page_size: u32,
    #[serde(default)]
    pub account_id: String,
    #[serde(default)]
    pub status: String,
}

/// 清空日志请求。
#[derive(Debug, Deserialize)]
pub struct PublishLogClearRequest {
    pub owner_id: i64,
    /// 保留最近 N 天；0 表示清空全部。
    #[serde(default)]
    pub days: u32,
}

/// 分页查询发布日志。
#[tauri::command]
pub fn publish_log_list(
    state: State<'_, PublishLogHandle>,
    request: PublishLogListRequest,
) -> common::OpenDeskResult<(Vec<PublishLog>, u32)> {
    let service = PublishLogService::new(state.store.as_ref());
    let query = PublishLogQuery {
        page: request.page,
        page_size: request.page_size,
        account_id: request.account_id,
        status: request.status,
    };
    service.list(request.owner_id, &query)
}

/// 清空早于指定天数的发布日志。
#[tauri::command]
pub fn publish_log_clear(
    state: State<'_, PublishLogHandle>,
    request: PublishLogClearRequest,
) -> common::OpenDeskResult<()> {
    let service = PublishLogService::new(state.store.as_ref());
    service.clear_older_than(request.owner_id, request.days)
}
